### DISCUSSION TITLE
No download anymore from any site

### DISCUSSION THREAD

**Discussion by mocos46** [aclap-dev/video-downloadhelper#20] [Help and Q&A]
### Your question

I constantly get an error from any site. My configuration is:
version: 8.1.4.0
beta: false
build date: 2024-01-04
build options: linuxlic: false, noyt: false, browser: firefox
lang: it
license: accepted
key (only first 16 fisrt characters): 4d57b04ec1da46c0
platform: x86-64 mac
UA: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.14; rv:109.0) Gecko/20100101 Firefox/115.0
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.10",
  "binary": "/Applications/net.downloadhelper.coapp.app/Contents/MacOS/vdhcoapp",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "mac",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "/var/root",
  "converterError": "Exit without answer"
}

**Comment by paulrouget** [comment]
- What version of MacOS do you use?
- Open the CoApp from your application folder. Do you see a notification?

Can you run this command in a terminal:

```
/Applications/net.downloadhelper.coapp.app/Contents/MacOS/vdhcoapp --info
```

And tell me what it says?

**Reply by mocos46** [reply]
Thanks for your answer. Mojave (10.14.6) - Nothing happens double-clicking net.downloadhelper.coapp
the terminal command gets:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.10",
  "binary": "/Applications/net.downloadhelper.coapp.app/Contents/MacOS/vdhcoapp",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "mac",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "/var/root",
  "converterError": "Exit without answer"
}

**Comment by paulrouget** [comment]
> Mojave (10.14.6) 

Ah. We don't support old versions of MacOS for now. That's why you're having this issue.

Good news is, I'm working on a fix for that! Please be patient. Hope to have something soon.

**Reply by mocos46** [reply]
Thank you very much for your answer. And for your efforts! I cannot afford to spend 3.5K euros for a new Mac and 14K euros for new software… Good works!

**Reply by paulrouget** [reply]
Hi! I'm trying to make a version of the app for old MacOS. But I can't test as I only have a newer Mac. Can you test for me? I'm honestly not sure it's gonna work, but let's see: https://github.com/aclap-dev/vdhcoapp/releases/latest/download/vdhcoapp-mac13-x86_64-installer.pkg

**Reply by mac-arthur1** [reply]
hi paul  your on the right track  Video DownloadHelper
Version 7.6.6 was able to see 
Found companion app: VdhCoApp 2.0.15
Companion app binary: /Applications/net.downloadhelper.coapp.app/Contents/MacOS/vdhcoapp
able to reconize my key

but it does not download the few Youtubes I tried
it did download a Vimeo but could not play it
but if I downloaded and converted it would play that 

I suspect I can find some other sites in might download

but it used to download Youtube  - which is a lot of stuff !

I can't test or play more tonight - its late, work tomorrow
and the soonest I can try more will be tomorrow night or over 
weekend  -- don't have the old mac at work.

fyi  I do have a newer mac, but for ergonomic reasons I don't 
use it so much  --- but 8/2 worked , and I think also on my PC at work, but 
I don't download and surf there.

THANKS, so far  ---
<img width="506" alt="Screen Shot 2024-02-01 at 9 59 21 PM" src="https://github.com/aclap-dev/video-downloadhelper/assets/158183178/c347200e-7d77-4712-bc3d-1b3cad789204">

**Reply by paulrouget** [reply]
Glad to hear things are better now!

Version 7.6.6 of the addon is too old for that new app.

Please install a more recent version: https://github.com/aclap-dev/video-downloadhelper/wiki#installation

**Reply by GrampaWildWilly** [reply]
Let me get some clarifying detail here.  Are you saying that even on an antique Mac, you could be running VDH 8.2.0.18a1 beta (the bleeding edge) as long as you're running a CoApp targeted at the older Mac platform?

**Reply by paulrouget** [reply]
Yes.

**Reply by mac-arthur1** [reply]
I'm not at the "old mac" now but I tired briefly last pm  - used 8.0 and 1.6  co ap - it did now download the Youtube that it would not before,  but very slow  - not like in past.  And I too wondered wasn't the idea that ext 8+ did not work well or maybe at all on the older Macs.  Unless your tweaking in the background, and are eliminating bugs that might have been problems  ?  Or is the issue only the Coapp ?  Can give 8.1 a try tonight.  And does work, just maybe slow ?  THANKS  !!

**Reply by GrampaWildWilly** [reply]
The 8.x browser extension requires a 2.x CoApp.  But the good news is you can now use the very latest beta, which has some significant bug fixes, specifically the high-resolution mkv variants on YouTube that used to fail, with your specially constructed CoApp for older releases of Mac.  For the browser extension (VDH itself) go here:

https://github.com/aclap-dev/video-downloadhelper/wiki/Beta-versions

For your special version of the CoApp for old Macs, go to the link Paul posted upthread here:

https://github.com/aclap-dev/vdhcoapp/releases/latest/download/vdhcoapp-mac13-x86_64-installer.pkg

For the speed thing you just mentioned, that is due to the fact that you were using a mismatched configuration: VDH 8.0 with an old CoApp 1.6.3.  On that Mac, which I gather is actually a new Mac, you should be able to use the latest VDH 8.x beta from the link I just provided.  And since you're talking about a new Mac, I believe this link for the CoApp should work:

https://github.com/aclap-dev/video-downloadhelper/wiki/CoApp-Installation

It doesn't look like it but that page has links to beta versions of the CoApp, something introduced in this discussion:

https://github.com/aclap-dev/video-downloadhelper/discussions/6

You just need to be careful to download the CoApp for the modern Mac onto your modern Mac, & the CoApp for the older Mac onto your older Mac.  But you can now use the latest VDH beta, which happens to be 8.2.0.18a1 beta at this moment (could be another new one tomorrow, who knows?) on both your old Mac & your new Mac.  I encourage you to use this updated VDH beta on both your systems because it does correct a number of problems.  Also, using this latest beta with the right CoApp on whichever machine should correct that speed problem you observed.

**Reply by mac-arthur1** [reply]
Looks good  - on my "old Mac"  (and no was not doing this on my "newer Mac" - just commented that I had one )
Video DownloadHelper  Version 8.2.0.18a1 (beta) + companion app: VdhCoApp 2.0.15
not sure if its absolutely as fast as it was, but its fine --

I wonder what the back end issues were - in simple terms, not developer language.  - just a buggy first attempt at the roll back scripts ?    

well thanks all !

**Reply by GrampaWildWilly** [reply]
It is my understanding that there are certain features of the newer versions of the operating system that the CoApp happens to use.  But there are other ways of accomplishing the same things (or close enough things to get by) if you restrict yourself to the feature set on the older release.  This was the explanation I got from Paul when he was working on a special version of the CoApp for Windows 7.  I was on W7 until just the beginning of December when I finally got W11 installed & working on a new computer.  Prior to that, Paul created several versions of a W7 CoApp that I tried & finally, he stomped on all the bugs & it worked.  I thanked him by turning around & upgrading to W11.  Isn't that special?  So I would imagine it's the same idea for the various versions of Mac.

As for speed, there's so many variables involved that it's hard to properly even discuss the topic.  There's the speed of service offered by the serving web site.  That can vary based on the load their users are demanding.  Then there's the speed of the connection your ISP is giving you.  That's supposed to be relatively constant but . . . we all know how that goes.  Then there's the Internet itself.  The routings & servers you go through can change based on network load, & there's nothing you can do about that.  It's just the way it is.  Under Windows, we have something called the Resource Monitor.  It looks like this:

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/a4f54025-61a9-489c-ba21-c167f1b7617c)

I have to believe you have a tool on Mac that shows you information like this.  Next time you're downloading something with VDH, look at that tool & see just what is going on with your network connection.  Do this for a number of downloads over a span of, let's say, a week.  That should give you a feel for what is normal for your system.  If you still think there is a speed issue, you should open a new discussion here & we can discuss it.  Gee.  Discussing a topic in a discussion.  What a concept!

It's great that you are running again.  You should identify one post here as the answer.  Paul already closed this discussion but it's still marked as unanswered, which is weird.  So pick an appropriate post & call it the answer.

**Comment by paulrouget** [comment]
I'm gonna close this as the main issue is fixed.

If this doesn't work for you still, please open a new ticket :)

Thank you!
